### ISSUE TITLE
Incorrect parsing of floats/doubles

### ISSUE THREAD

**Issue by LeStegii** [vincentrussell/java-ini-parser#4]
Hello :)
When using the ini-parser to load floats/doubles from a file, it will read X.0 as X0.0.
For example when reading "1.0", the value you get is "10.0" as the dot seems to be ignored.

Example config file with floats/doubles
```ini
[example]
boolean = true
double = 1.0
float = 1.0
integer = 1
long = 1
string = modified
enum = ANOTHER_EXAMPLE
```

Example code for loading the float
```java
public void test() {
        Ini ini = new Ini();
        try {
            ini.load(new File("src/test/resources/example.ini"));
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
        System.out.println(ini.getValue("example", "float", Float.class));
    }
```

It also doesn't matter whether you cast the value or not.

Is this expected behaviour or a bug?

**Comment by vincentrussell** [comment]
definitely seems like I bug.  I"ll try to take a look tomorrow.

**Comment by LeStegii** [comment]
Thank you :)
It works if you just leave out the dot by the way.

```
double = 1
float = 1
```

Printing the value results in `1.0` then.

**Comment by vincentrussell** [comment]
I wrote the following test:

    @Test
    public void floattest() throws IOException {
        Ini ini = new Ini();
        ini.load(Thread.currentThread().getContextClassLoader().getResourceAsStream("samples/floattest.ini"));
        assertEquals((float)1.0, (float)ini.getValue("example", "float", Float.class), 0);
        assertEquals((double)1.0, (double)ini.getValue("example", "double", Double.class), 0);
    }
}

I can't duplicate your results.

What operating system and jdk are you using?

**Comment by LeStegii** [comment]
Sorry for the late answer, I'm using Windows 10 with OpenJDK 8.
```
openjdk version "1.8.0_352"
OpenJDK Runtime Environment (Temurin)(build 1.8.0_352-b08)
OpenJDK 64-Bit Server VM (Temurin)(build 25.352-b08, mixed mode)
```

**Comment by vincentrussell** [comment]
I don't have Windows, but I've tried to duplicate your results with multiple JDKs and the test that i wrote passes every time:
`    @Test
    public void floattest() throws IOException {
        Ini ini = new Ini();
        ini.load(Thread.currentThread().getContextClassLoader().getResourceAsStream("samples/floattest.ini"));
        assertEquals((float)1.0, (float)ini.getValue("example", "float", Float.class), 0);
        assertEquals((double)1.0, (double)ini.getValue("example", "double", Double.class), 0);
    }`

**Comment by vincentrussell** [comment]
```
    @Test
    public void floattest() throws IOException {
        Ini ini = new Ini();
        ini.load(Thread.currentThread().getContextClassLoader().getResourceAsStream("samples/floattest.ini"));
        assertEquals((float)1.0, (float)ini.getValue("example", "float", Float.class), 0);
        assertEquals((double)1.0, (double)ini.getValue("example", "double", Double.class), 0);
    }
```

**Comment by LeStegii** [comment]
I just tried cloning the repo and building it for myself ([logs](https://paste.feed-the-beast.com/view/raw/f69cda06)).
The build failed due to failing tests, for example 3.14 was parsed as 314 again.

**Comment by vincentrussell** [comment]
I'm sorry.  I was finally able to test this with windows 10 and the test that I wrote passes.   Only one test failed due to CRLF issues, orderIsMaintained, which I fixed with: 79c170829bbbcef8a97b2381f8d30773e941fd76


I'm going to have to close this ticket.    This is the environment that I used:
$ mvn -version
Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
Maven home: C:\Users\vince\Apps\apache-maven-3.9.1
Java version: 1.8.0_362-362, vendor: OpenLogic-OpenJDK, runtime: C:\Users\vince\Apps\openlogic-openjdk-8u362-b09-windows-64\jre
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
